### ISSUE TITLE
Axis flips when indexing an high-dimensional Array

### ISSUE THREAD

**Issue by Xin-yang-Liu** [google/jax#15653]
### Description

To reproduce
```python
import jax.numpy as jnp

a=jnp.ones((5,2,10,20))
b=[0,4,8] #the index
print(a[0,:,b].shape)
print(a[:,:,b].shape)
```
output
```output
(3, 2, 20)
(5, 2, 3, 20)
```
While the expected output for the first print should be ```(2,3,20)```

### What jax/jaxlib version are you using?

jax v0.4.8; jaxlib v0.4.7+cuda11.cudnn86

### Which accelerator(s) are you using?

Tested on both CPU and GPU

### Additional system info

Google Colab, accelerator T4

### NVIDIA GPU info

```NVIDIA-SMI 525.85.12    Driver Version: 525.85.12    CUDA Version: 12.0```

**Comment by hawkinsp** [comment]
This is working as intended: these are the semantics of NumPy advanced indexing. See: https://numpy.org/doc/stable/user/basics.indexing.html#combining-advanced-and-basic-indexing

and note classic NumPy has exactly the same behavior:

```
In [26]: import numpy as np
    ...:
    ...: a=np.ones((5,2,10,20))
    ...: b=[0,4,8] #the index
    ...: print(a[0,:,b].shape)
    ...: print(a[:,:,b].shape)
(3, 2, 20)
(5, 2, 3, 20)
```

Hope that helps!

**Comment by jakevdp** [comment]
Thanks for the report! NumPy gives the same results, and JAX's indexing semantics are designed to match NumPy's, so it looks like this is working as intended:
```python
import numpy as np
a=np.ones((5,2,10,20))
b=[0,4,8] #the index
print(a[0,:,b].shape)
print(a[:,:,b].shape)
```
```
(3, 2, 20)
(5, 2, 3, 20)
```

**Comment by jakevdp** [comment]
@hawkinsp by a nose... 🏁 🏁
